### PR TITLE
new AWS module for ec2 dhcp option facts

### PR DESCRIPTION
**Pull request by naslanidis**
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME

ec2_vpc_dhcp_options_facts
##### ANSIBLE VERSION

```
ansible 2.2.0 (devel f39ad1f13a) last updated 2016/05/01 14:01:36 (GMT +1000)
config file = /etc/ansible/ansible.cfg
configured module search path = ['/usr/share/ansible']

```
##### SUMMARY

This module returns DHCP Option Set facts for an AWS account.

```
TASK [print output] ************************************************************
ok: [localhost] => {
    "msg": {
        "changed": false, 
        "dhcp_options": [
            {
                "dhcp_configurations": [
                    {
                        "key": "domain-name", 
                        "values": [
                            {
                                "value": "test.cloud.local"
                            }
                        ]
                    }, 
                    {
                        "key": "domain-name-servers", 
                        "values": [
                            {
                                "value": "192.168.1.1"
                            }, 
                            {
                                "value": "192.168.1.2"
                            }
                        ]
                    }, 
                    {
                        "key": "ntp-servers", 
                        "values": [
                            {
                                "value": "192.168.1.1"
                            }
                        ]
                    }
                ], 
                "dhcp_options_id": "dopt-5413a123", 
                "tags": [
                    {
                        "key": "Name", 
                        "value": "test-dhcp-set-1"
                    }, 
                    {
                        "key": "name", 
                        "value": "test-dhcp-set-1"
                    }
                ]
            }
        ]
    }
}

```

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @naslanidis for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/amazon/ec2_vpc_dhcp_options_facts.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+'''
+
+try:
+    import json
+    import botocore
+    import boto3   
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
```
Comment (gundalow) [inline cloud/amazon/ec2_vpc_dhcp_options_facts.py:87]: `HAS_LIBS` may be a better name as you are checking for multiple libraries

File: cloud/amazon/ec2_vpc_dhcp_options_facts.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+            filters = dict(type='dict', default=None, ),
+            DryRun = dict(type='bool', default=False),
+            DhcpOptionsIds = dict(type='list', default=None)
+        )
+    )
+
+    module = AnsibleModule(argument_spec=argument_spec)
+
+     # Validate Requirements
+    if not HAS_BOTO3:
```
Comment (gundalow) [inline cloud/amazon/ec2_vpc_dhcp_options_facts.py:148]: `HAS_LIBS`

File: cloud/amazon/ec2_vpc_dhcp_options_facts.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        module.fail_json(msg='json and botocore/boto3 is required.')
```
Comment (gundalow) [inline cloud/amazon/ec2_vpc_dhcp_options_facts.py:149]: One or more of json, botocore and boto3 are missing.

**Comment by naslanidis** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @naslanidis for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by ljudzia** [comment]
Tested all works thanks - shipit

**Comment by mmochan** [comment]
Thanks @naslanidis just what I was looking for. Works as described, and no issues

**Comment by gregdek** [comment]
Thanks @naslanidis for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by naslanidis** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @naslanidis for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by kalyat** [comment]
shipit

**Comment by gundalow** [comment]
shipit

**Comment by gregdek** [comment]
Thanks again to @naslanidis for this PR, and thanks @ansible/core for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gundalow** [comment]
new ec2 dhcp option facts module

**Comment by gundalow** [comment]
Thanks :)

**Pull request merged**